aws-sigv4/src/lib.rs

```diff
@@ -82,11 +67,21 @@ pub struct Config<'a> {
 }
 
 #[derive(Debug, PartialEq)]
+#[non_exhaustive]
 pub struct SigningSettings {
     /// We assume the URI will be encoded _once_ prior to transmission. Some services
     /// do not decode the path prior to checking the signature, requiring clients to actually
     /// _double-encode_ the URI in creating the canonical request in order to pass a signature check.
     pub uri_encoding: UriEncoding,
+
+    pub signed_body_header: SignedBodyHeaderType,
+}
+
+#[non_exhaustive]
+#[derive(Debug, Eq, PartialEq)]
+pub enum SignedBodyHeaderType {
```

Comment:
Bit of a mouthful... what's a `BodyHeader`?  Also, for enums, let's call it `Kind` instead of `Type` due to keyword reservations?

Reply:
yeah a nicer name would be good here. What about `PayloadChecksumHeaderKind`?